src/rpcrawtransaction.cpp

```diff
@@ -24,111 +23,280 @@ extern std::string GetTxProject(uint256 hash, int& out_blocknumber, int& out_blo
+                res.push_back(std::make_pair("Poll Question", sPollQuestion));
+                std::replace(sPollAnswers.begin(), sPollAnswers.end(), ';', ',');
+                res.push_back(std::make_pair("Poll Answers", sPollAnswers));
+                sPollShareType = GetShareType(std::stod(sPollShareType));
+                res.push_back(std::make_pair("Poll Share Type", sPollShareType));
+                res.push_back(std::make_pair("Poll URL", sPollUrl));
+                res.push_back(std::make_pair("Poll Duration", sPollDays + " days"));
+                res.push_back(std::make_pair("Poll Expires", TimestampToHRDate(std::stod(sPollExpiration))));
+            }
+
+            else if (sMessageType == "vote")
+            {
+                res.push_back(std::make_pair("Contract Type", "Vote"));
+
+                std::string sVoteTitle = ExtractXML(msg, "<TITLE>", "</TITLE>");
+                std::string sVoteShareType = GetPollXMLElementByPollTitle(sVoteTitle, "<SHARETYPE>", "</SHARETYPE>");
+                sVoteShareType = GetShareType(std::stod(sVoteShareType));
+                std::replace(sVoteTitle.begin(), sVoteTitle.end(), '_', ' ');
+                std::string sVoteAnswer = ExtractXML(msg, "<ANSWER>", "</ANSWER>");
+                std::replace(sVoteAnswer.begin(), sVoteAnswer.end(), ';', ',');
+
+                res.push_back(std::make_pair("Poll Title", sVoteTitle));
+                res.push_back(std::make_pair("Poll Answer", sVoteAnswer));
+                res.push_back(std::make_pair("Poll Share Type", sVoteShareType));
+            }
+
+            else if (sMessageType == "project")
+            {
+                res.push_back(std::make_pair("Contract Type", "Project"));
+
+                std::string sProjectName = ExtractXML(msg, "<MK>", "</MK>");
+                std::string sProjectURL = ExtractXML(msg, "<MV>", "</MV>");
+                std::string sProjectCommit = ExtractXML(msg, "<MA>", "</MA>");
+
+                if (sProjectCommit == "A")
+                    res.push_back(std::make_pair("Project Commit", "Add"));
+
+                else if (sProjectCommit == "D")
+                    res.push_back(std::make_pair("Project Commit", "Delete"));
+
+                else
+                    res.push_back(std::make_pair("Project Commit", "Unknown"));
+
+                res.push_back(std::make_pair("Project Name", sProjectName));
+                res.push_back(std::make_pair("Project URL", sProjectURL));
+            }
+            else
+            {
+                res.push_back(std::make_pair("Contract Type", "Unknown"));
+
+                if (fDebug)
+                    res.push_back(std::make_pair("Contract Data", msg));
+            }
+        }
+
+        else if (sTxMessage.length())
         {
-            res["IsSuperBlock"]=true;
+            res.push_back(std::make_pair("Message Type", "Transaction Message"));
+            res.push_back(std::make_pair("Message", sTxMessage));
```

Comment:
"Text"? "Message Text"?

Reply:
wasnt sure but was trying to separate things as best i can to not cause confusion will wait for ravons review on this as well

Reply:
-- complete